### PR TITLE
Use email address when customer Id not available

### PR DESCRIPTION
**Pull request by davidstanley01**
In #79, @miken32 reported that attempting to use an email address to create/look-up a customer was failing with a reported error of `"Return value of HelpScout\Api\Customers\Customer::getId() must be of the type integer, null returned"`. Given the [docs](https://developer.helpscout.com/mailbox-api/endpoints/conversations/create/) say that either is fine, this is an issue.

After some investigation, I found 2 issues. First, the return type for the `Customer::getId()` method did not allow for a possible `null` value. Second, when attempting to set the customer data on a conversation or a thread, we were not accounting for the possibility of only having an email address.

I updated the `getId` method return type to allow for a `null` value. Then, I updated the logic to consider email address when assigning the customer values to the `Conversation` and any thread that uses the `HasCustomer` trait. Finally, I added a convenience method to get the first email address associated with a customer.

### CODE REVIEW

File: src/Customers/Customer.php [hunk 1]
```diff
@@ -426,6 +426,17 @@ public function getEmails(): Collection
...
 
+    /**
+     * @return null|string
+     */
+    public function getFirstEmail(): ?string
+    {
+        $emails = $this->emails->toArray();
+        $email = array_shift($emails);
+
+        return $email ?? null;
```
Comment (miken32) [inline src/Customers/Customer.php]: The `toArray()` method returns an array of `Email` objects. If that behaviour is expected, this should be `return $email->value ?? null;`

**Comment by dacopavlov** [comment]
Guys any new about this issue, i mean when do your plan to deploy the fix? 
Or if there is an other way to get a all conversations from given email address please if its possible provide how to with an example.

Thanks.

**Comment by davidstanley01** [comment]
@dacopavlov ​This is going to go out today once I make a change to address the first comment on that PR. My apologies for the delay.

File: src/Customers/Customer.php (hunk 1 above)
Comment (davidstanley01) [reply src/Customers/Customer.php]: @miken32 I've updated this. The `ConversationTest` was passing a string as an email when creating the collection to test the `extract` method. I'm fairly sure this is where the simply `$email ?? null` check came from. 

After updating that test to pass an `Email` entry object into the collection, it became apparent what the solution should be in the `Customer::getFirstEmail` method. 

Thanks for pointing this out.

**Comment by davidstanley01** [comment]
Looks like this is showing the commits from #82 in here as well...

**Pull request merged**